bms-monitor.js

```diff
@@ -1,18 +1,27 @@
-const {expect} = require('chai');
+const { expect } = require('chai');
 
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
-        console.log('Temperature is out of range!');
-        return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
-        console.log('Charge rate is out of range!');
-        return false;
-    }
-    return true;
+//To check if temp in range,returns true if it passes
+function temperatureInRange(inputTemperature) {
+    return (inputTemperature > 0 && inputTemperature < 45) ? true : false;
+

```

Comment:
The pattern of code in this function and the following two are same. Bring out the common pattern into an abstraction (function), which will determine if a given parameter is within the given range. Then you can pass different parameters and ranges to it